### PR TITLE
Name sub_<hex>() functions

### PR DESCRIPTION
**Pull request by rwjuk**
* [ ] 14937 `sub_98197C`
* [ ] 2534 `sub_98196C`
* [ ] 1240 `sub_98199C`
* [x] ~~77 `sub_6C84CE`~~ - `window_ride_construction_update_active_elements()`
* [x] ~~54 `sub_65E72D`~~ - `get_num_of_sheltered_eighths()`
* [x] ~~52 `sub_6C9627`~~ - `ride_construction_invalidate_current_track()`
* [x] ~~44 `sub_693B58`~~ - `peep_update_current_action_sprite_type()`
* [ ] 17 `sub_6C683D`
* [ ] 16 `sub_6CA2DF`
* [x] ~~15 `sub_693C9E`~~ - `peep_perform_next_action()`
* [x] ~~13 `sub_6EA934`~~ - `window_fits_within_space()`
* [x] ~~13 `sub_6D01B3`~~ - `place_virtual_track()`
* [x] ~~11 `sub_6E7FF3`~~ - `viewport_redraw_after_shift()`
* [x] ~~9 `sub_6CC3FB`~~ - `ride_initialise_construction_window()`
* [x] ~~9 `sub_6A759F`~~ - `footpath_update_queue_chains()`
* [x] ~~8 `sub_6C96C0`~~ - `ride_construction_remove_ghosts()`
* [x] ~~8 `sub_693BE5`~~ - `peep_switch_to_special_sprite()`
* [x] ~~8 `sub_693BAB`~~ - `peep_switch_to_next_action_sprite_type()`
* [ ] 7 `sub_6CB945`
* [x] ~~7 `sub_6CA162`~~ - `place_provisional_track_piece()`
* [ ] 7 `sub_68A15E`
* [x] ~~7 `sub_684AC3`~~ - `research_reset_current_item()`
* [x] ~~6 `sub_88ABA4`~~ - `paint_crooked_house_structure()`
* [ ] 5 `sub_98198C`
* [x] ~~5 `sub_6ED990`~~ - `set_cursor()`
* [x] ~~5 `sub_6EA8EC`~~
* [x] ~~5 `sub_6DFED0`~~ - `reset_researched_scenery_items()`, see #5536
* [x] ~~5 `sub_69AF1E`~~ - `peep_decide_and_buy_item()`
* [ ] 5 `sub_68862C`
* [x] ~~4 `sub_6EA95D`~~
* [ ] 4 `sub_6D63D4`
* [ ] 4 `sub_6C94D8`
* [x] ~~4 `sub_6A7642`~~ - `footpath_update_queue_entrance_banner()`
* [ ] 4 `sub_68B3FB`
* [ ] 4 `sub_68B089`
* [x] ~~4 `sub_689174`~~ - `viewport_adjust_for_map_height()`
* [x] ~~3 `sub_6EA73F`~~
* [ ] 3 `sub_6E1F34`
* [x] ~~3 `sub_6DB807`~~ - `vehicle_update_reverser_car_bogies()`
* [ ] 3 `sub_6D7AC0`
* [ ] 3 `sub_6CBCE2`
* [ ] 3 `sub_6BC2F3`
* [x] ~~3 `sub_6B7588`~~ - `peep_update_ride_inspected()`
* [ ] 3 `sub_6B5952`
* [ ] 3 `sub_6AB211`
* [ ] 3 `sub_6A3F61`
* [x] ~~3 `sub_69ED9E`~~ - `count_free_misc_sprite_slots()`
* [x] ~~3 `sub_68F8CD`~~ - `peep_decide_whether_to_leave_park()`
* [ ] 3 `sub_68F41A` - see comment below, should probably be folded into `peep_update_all()`
* [ ] 3 `sub_68B2B7`
* [x] ~~3 `sub_685EBC`~~ - `paint_floating_money_effect()`
* [ ] 3 `sub_68371D`
* [ ] 3 `sub_679236`
* [ ] 3 `sub_679074`
* [ ] 3 `sub_679023`
* [x] ~~3 `sub_666EEF`~~ - `place_park_entrance_get_map_position()`
* [ ] 2 `sub_6DF21B`
* [ ] 2 `sub_6DF01A`
* [x] ~~2 `sub_6DEDE8`~~ - `vehicle_update_handle_scenery_door()`
* [ ] 2 `sub_6DBF3E`
* [x] ~~2 `sub_6DB7D6`~~ - `vehicle_reverse_reverser_car()`
* [x] ~~2 `sub_6D2189`~~ - `track_design_place_preview()`
* [x] ~~2 `sub_6B2FA9`~~ - `close_ride_window_for_construction()`
* [ ] 2 `sub_6A4101`
* [x] ~~2 `sub_688956`~~ - `call_event_unknown_14_on_all_windows()`
* [ ] 2 `sub_688217` - doesn't appear to exist, used in lightfx.c L294
* [ ] 2 `sub_67933B`
* [x] ~~2 `sub_66F6E3`~~ - elided
* [x] ~~1 `sub_6EDE88`~~ - declared but never defined
* [x] ~~1 `sub_6A8ACF`~~ - comment only
* [x] ~~1 `sub_68DABD`~~ - comment only 
* [x] ~~1 `sub_66E4BA`~~ - comment only
* [x] ~~1 `sub_65E621`~~ - comment only

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Let me know when you're done. Or is this a WIP? In that case, please put the label on it.

**Comment by rwjuk** [comment]
Whoops! Forgot that. I was planning to just go through all of them (happy for others to push to this branch too if they want).

File: src/openrct2/ride/ride_ratings.c [hunk 1]
```diff
@@ -904,10 +904,10 @@ static uint32 ride_ratings_get_proximity_score()
 }
 
 /**
- * Seems to calculate how much of the track is sheltered in eighths.
+ * Calculates how much of the track is sheltered in eighths.
  *  rct2: 0x0065E72D
  */
-static sint32 sub_65E72D(rct_ride *ride)
+static sint32 get_num_of_sheltered_eighths(rct_ride *ride)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/ride_ratings.c]: See https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/ride.h#L225
I've called this `undercover_portion` but perhaps a rename to `sheltered_eigths` would be better.

File: src/openrct2/ride/ride_ratings.c [hunk 2]
```diff
@@ -904,10 +904,10 @@ static uint32 ride_ratings_get_proximity_score()
 }
 
 /**
- * Seems to calculate how much of the track is sheltered in eighths.
+ * Calculates how much of the track is sheltered in eighths.
```
Comment (duncanspumpkin) [inline src/openrct2/ride/ride_ratings.c]: I think this function needs refactoring for its outputs. The `sint32` is comprised of two `uint8`'s one of which is the sheltered eights and the other is the sheltered eights or 7 depending on if the ride is a covered ride.

Comment (rwjuk) [inline src/openrct2/ride/ride_ratings.c]: Probably best to put that in a separate PR - I don't disagree though.

File: src/openrct2/ride/ride_ratings.c (hunk 1 above)
Comment (rwjuk) [inline src/openrct2/ride/ride_ratings.c]: I'll change that too for consistency - I agree that sheltered eighths is more accurate.

**Comment by rwjuk** [comment]
We've also got some `loc_<hex>()` functions to name too.

File: src/openrct2/windows/editor_bottom_toolbar.c [hunk 3]
```diff
@@ -271,7 +253,17 @@ void window_editor_bottom_toolbar_jump_forward_from_object_selection()
 		return;
 
 	if (gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER) {
-		sub_66F6E3();
+		for (sint32 i = 0; i < 4; i++) {
```
Comment (Gymnasiast) [inline src/openrct2/windows/editor_bottom_toolbar.c]: I think this line (up to and including 262) should be in a function (if one doesn't exist already). That function should not be in this window, but in the regular research file. (This can be put into a separate issue, of course.)

Comment (rwjuk) [inline src/openrct2/windows/editor_bottom_toolbar.c]: Fair point - no need for a separate issue though as I've done some refactoring of research already. Copy your comment to #5536?

Comment (rwjuk) [inline src/openrct2/windows/editor_bottom_toolbar.c]: It might be possible to rewrite this using existing research functions anyway (e.g. one I just created, I think).

File: src/openrct2/ride/ride.h [hunk 4]
```diff
@@ -219,9 +219,9 @@ typedef struct rct_ride {
 	union {
 		uint8 inversions;			// 0x114 (???X XXXX)
 		uint8 holes;				// 0x114 (???X XXXX)
-		// The undercover portion is a very rough approximation of how much of the ride is undercover.
+		// This a very rough approximation of how much of the ride is undercover.
```
Comment (pizza2004) [inline src/openrct2/ride/ride.h]: Your auto replacement seems to have just deleted " undercover portion " instead of replacing it with " sheltered eighths ".  If you meant it to be a this after all, you need to add an is afterwords.

Comment (rwjuk) [inline src/openrct2/ride/ride.h]: It was intentional, but I agree it could probably be clearer.

**Comment by rwjuk** [comment]
IMO, `sub_68F41A()` should be folded into `peep_update_all()`. That's the only place it's called from (and given the index parameter used to determine how often to update certain things, the only place it really makes sense to call it from). Thoughts?

File: src/openrct2/ride/ride.h (hunk 4 above)
Comment (Gymnasiast) [inline src/openrct2/ride/ride.h]: "This a" is not proper English. It should be "This is a". (Which is also what pizza2004 said.)

Comment (rwjuk) [inline src/openrct2/ride/ride.h]: Ohhh, I see it now. Yeah, will fix.

**Comment by janisozaur** [comment]
@rwjuk I don't think `sub_68F41A` should be simply _folded_ into `peep_update_all`. It is a rather large and complex function.

File: src/openrct2/windows/ride_construction.c [hunk 5]
```diff
@@ -539,14 +531,22 @@ static sint32 ride_get_alternative_type(rct_ride *ride)
 		ride->type;
 }
 
+/* move to ride.c */
+static void close_ride_window_for_construction(rct_windownumber number)
+{
+	rct_window* w = window_find_by_number(WC_RIDE, number);
+	if (w != NULL && w->page == 1)
```
Comment (rwjuk) [inline src/openrct2/windows/ride_construction.c]: Not honestly sure why the ride window needs to be closed before construction, but only if it's on the vehicle tab...

**Comment by rwjuk** [comment]
I think I've got as many of these as I feel confident naming at the moment. If anyone else wants to name any, they're welcome to push to the branch, otherwise this can be merged as is, and we can keep track of the remaining nameless functions on the issue (#5525).

File: src/openrct2/windows/ride_construction.c (hunk 5 above)
Comment (Gymnasiast) [inline src/openrct2/windows/ride_construction.c]: It might be because building can have a direct effect on the number of trains and cars per train. Too much interference.

File: src/openrct2/interface/viewport.c [hunk 6]
```diff
@@ -250,7 +250,7 @@ void sub_689174(sint16* x, sint16* y, sint16 *z)
 	*z = height;
 }
 
-static void sub_6E7FF3(rct_drawpixelinfo *dpi, rct_window *window, rct_viewport *viewport, sint32 x, sint32 y)
+static void viewport_redraw_after_shift(rct_drawpixelinfo *dpi, rct_window *window, rct_viewport *viewport, sint32 x, sint32 y)
```
Comment (janisozaur) [inline src/openrct2/interface/viewport.c]: Add a comment above

File: src/openrct2/peep/peep.c [hunk 7]
```diff
@@ -1392,12 +1392,14 @@ static uint8 peep_get_action_sprite_type(rct_peep* peep)
 	}
 }
 
-void sub_693B58(rct_peep* peep){
+void peep_update_current_action_sprite_type(rct_peep* peep){
```
Comment (janisozaur) [inline src/openrct2/peep/peep.c]: Add a comment above:
```
/*
 *  rct2: 0x00693B58
 */
```
Mind the double space.

File: src/openrct2/ride/ride.c [hunk 8]
```diff
@@ -1401,7 +1400,7 @@ void sub_6C96C0()
 	}
 }
 
-void sub_6C9627()
+void ride_construction_invalidate_current_track()
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: Add a comment above

File: src/openrct2/windows/ride_construction.c (hunk 5 above)
Comment (rwjuk) [inline src/openrct2/windows/ride_construction.c]: Ahh, that makes sense.

**Comment by rwjuk** [comment]
@Gymnasiast I thought it'd be better done this way rather than having several PRs for the same thing - easier to keep track and such. Noted for the future.

File: src/openrct2/peep/peep.c (hunk 7 above)
Comment (IntelOrca) [inline src/openrct2/peep/peep.c]: We don't need to write original addresses now. The git history is good enough should we ever need to know what they were.

Comment (Gymnasiast) [inline src/openrct2/peep/peep.c]: Hm, are we sure it's not worth keeping these easy notes in case we messed up decompilation? I don't have a strong opinion on this in any case.

Comment (janisozaur) [inline src/openrct2/peep/peep.c]: I find it easier to know that a given function was disassembled and whether any bugs found in it could have been original. Without such comment I can still dig this information up, but it takes much longer.

Comment (rwjuk) [inline src/openrct2/peep/peep.c]: I put them in - I agree that it's helpful to know whether a function could have original bugs.

File: src/openrct2/peep/peep.c [hunk 9]
```diff
@@ -7841,7 +7847,7 @@ void peep_set_map_tooltip(rct_peep *peep)
 }
 
 
-void sub_693BAB(rct_peep* peep) {
+void peep_switch_to_next_action_sprite_type(rct_peep* peep) {
```
Comment (janisozaur) [inline src/openrct2/peep/peep.c]: Missed a spot
